src/components/TeamBuilderPage/TeamBuilderPage.jsx

```diff
@@ -0,0 +1,176 @@
+  const tiers = alliance.levels.map((level,i) => level.unitcount - (i > 0 ? alliance.levels[i-1].unitcount : 0));
+  let c = count;
+  return(
+    <div className={styles.activeAllianceContainerOuter}>
+      <div className={styles.activeAllianceContainer} style={{backgroundColor: `rgba(${alliance.color.split(' ').join(',')},.22)`}}>
+        <div className={styles.allianceImgContainer}>
+          <img className={styles.allianceImg} alt={synergy} src={`${process.env.PUBLIC_URL}/images/alliances/${synergy}.jpg`} />
+        </div>
+        <div className={styles.allianceTiersContainer}>
+          {
+            tiers.map(tier => {
+              return(
+                <div className={styles.allianceTier}>
+                  {
+                    [...Array(tier)].map(_ => {
+                      c = c - 1;
+                      return(
+                        <div className={`${styles.tierCount} ${c >= 0 ? styles.activeCount : null}`}></div>
+                      )
+                    })
+                  }
+                </div>    
+              )
+            })
+          }            
+        </div>
+      </div>
+    {level ? <div className={styles.synergyDescription}>{`(${levelUnitCount}) ${underlordsLoc[`dac_synergy_desc_${synergy}_${level}`]}`}</div> : null}
+  </div>
+  )
+}
+
+function transformName(str) {
+  const splitStr = str.toLowerCase().split('_');
+  for (let i = 0; i < splitStr.length; i++) {
+      splitStr[i] = splitStr[i].charAt(0).toUpperCase() + splitStr[i].substring(1);     
+  }
+  return splitStr.join(' '); 
+}
+
+export default class TeamBuilderPage extends React.Component {
+
+  state = {
+    searchValue: "",
+    team: [],
+    synergies: {},
+  }
+
+  computeAlliances = () => {
+    const team = [...new Set(this.state.team)]
+    const synergies = {};
+    Object.keys(alliances).map(a => alliances[a]).forEach(alliance => {
+      alliance.heroes.forEach(allianceHero => {
+        team.forEach(teamMember => {
+            if(allianceHero.key === teamMember) {
+              if(synergies[alliance.key]) {
+                synergies[alliance.key].count = synergies[alliance.key].count + 1
+              } else {
+                synergies[alliance.key] = {count: 1, level: 0}
+              }
+              alliance.levels.forEach((level, index) => {
+                if(synergies[alliance.key].count >= level.unitcount) {
+                  synergies[alliance.key].level = index + 1;
+                  synergies[alliance.key].levelUnitCount = level.unitcount;
+                }
+              })       
+            }
+        })
+      })
+    })
+    this.setState({synergies});
+  }
+
+  handleSearchChange = (e) => {
+    this.setState({searchValue: e.target.value})
+  }
+
+  handleHeroSelection = h => () => {
+    if(this.state.team.length < 10) {
+      this.setState({team: [...this.state.team, h.key]}, this.computeAlliances)
+    }
+    this.resetSearch();
+  }
+
+  handleHeroDeSelection = i => () => {
```

Comment:
extra level of function currying?

Reply:
I'm not sure if I understand. The onClick attribute is a function?

Reply:
doesn't it work with just (i) => { } ?

Reply:
yes but then we would have to write `onClick={() => this.handleHeroDeSelection(i)}` instead of just `onClick={this.handleHeroDeSelection(i)}` . Or I'm missing something

Reply:
I actually prefer the first one and like to keep the class functions only one level deep, but I'm fine with it this way.